client/src/main/java/building/FleetDropController.java

```diff
@@ -0,0 +1,175 @@
+            theShip.addEventHandler(MouseEvent.MOUSE_ENTERED, makeShadowWhenMoveOver);
+            theShip.addEventHandler(MouseEvent.DRAG_DETECTED, dragDetected);
+            theShip.addEventHandler(MouseEvent.MOUSE_EXITED, event -> theShip.setEffect(null));
+            theShip.setOnDragDone(event -> {
+                theShip.setFill(Color.BLACK);
+                event.consume();
+            });
+        }
+    }
+
+    EventHandler<DragEvent> seaFieldToShipWhenDraggedDone =
+            new EventHandler<DragEvent>() {
+                @Override
+                public void handle(DragEvent event) {
+                    SeaField field = (SeaField) event.getSource();
+                    field.marked();
+                    Integer column = field.getColumn();
+                    Integer row = field.getRow();
+
+                    Dragboard db = event.getDragboard();
+                    boolean success = false;
+                    if (db.hasString()) {
+                        success = true;
+                    }
+                    int shipLength = (int) ((Rectangle) event.getGestureSource()).getHeight() / FIELD_SIZE;
+                    Mast mast = new Mast(column, row);
+
+                    shipBoard.getChildren().add(mast);
+                    GridPane.setConstraints(mast, column, row);
+
+                    buildShip.removeEventHandler(MouseEvent.DRAG_DETECTED, dragDetected);
+                    buildShip.removeEventHandler(MouseEvent.MOUSE_ENTERED, makeShadowWhenMoveOver);
+                    fleet.startToBuildOneShip(mast, shipLength);
+
+                    event.setDropCompleted(success);
+                    event.consume();
+                    buildShip.setOpacity(0.2);
+                    if (buildShip.getHeight() / FIELD_SIZE > 1) port.setDisable(true);
+                }
+            };
+
+    EventHandler<DragEvent> seaFieldAcceptEventDraggedOver =
+            event -> {
+                Dragboard db = event.getDragboard();
+                if (db.hasString()) {
+                    event.acceptTransferModes(TransferMode.COPY);
+                }
+                event.consume();
+            };
+
+    EventHandler<DragEvent> changeColorwhenDragEntered =
+            event -> {
+                SeaField field = (SeaField) event.getSource();
+                field.setFill(Color.RED);
+                event.consume();
+            };
+
+    EventHandler<DragEvent> resetColorWhenDragExited =
+            event -> {
+                SeaField field = (SeaField) event.getSource();
+                field.reset();
+                event.consume();
+            };
+
+    EventHandler<MouseEvent> dragDetected =
+            event -> {
+                buildShip = (Rectangle) event.getSource();
+                Dragboard db = buildShip.startDragAndDrop(TransferMode.COPY);
+                ClipboardContent content = new ClipboardContent();
+                content.putString("Hello!");
```

Comment:
it some kind of test string?

Reply:
I change this string to "Ship", I'm not sure why, but this string must be set for content to work.